enhancements/baremetal-config.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Baremetal IPI Config Enhancement
+authors:
+  - "@sadasu"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-10-23
+last-updated: 2019-10-23
+status: implemented
+see-also:
+  - "https://github.com/openshift/api/pull/480"  
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+None.
+
+## Summary
+
+The configuration required to bring up a Baremetal IPI (metal3) cluster needs
+to be availabe in a Config Resource. The installer aleady adds some configuration
+to the Infrastructure Config CR. It is also aware of the additional config
+required during the deployment of a metal3 cluster. This enhancement proposes
+to extend the BareMetaPlatformStatus CR to hold these additional values needed
+for a metal3 deployment.
+
+## Motivation
+
+Adding these additional config items to the BareMetalPlatformStatus CR, allows
+versioning of these config items.
+
+### Goals
+
+The goal of this enhancement is to enable the installer to add vital config
+values for a Baremetal IPI deployment to a CR. The MAO will then be able to
+access these values when it kicks off a metal3 cluster.
+
+### Non-Goals
+
+All config values except the RHCOSImageUrl are not expected to change after a
+metal3 cluster is up. Handling changes to this value after install time has
+not been explored and that is a non-goal at this time.
```

Comment:
We could mention that the update lifecycle for bootimage is an unsolved problem for all platforms ref https://github.com/openshift/os/issues/381

Reply:
Done.

Reply:
@imcleod assembled some notes on this issue from a discussion we had a couple months ago as well: https://docs.google.com/document/d/1W_N3pcOuo6E_DcM5GQ731cTTN2VJ1KUa2A49_Zi3IRU/edit

Reply:
@ericavonb & @imcleod isit OK for me to add a link to this doc in my enhancement request?